docs/guide/testing.md

```diff
@@ -233,3 +233,42 @@ To use component previews:
 # config/application.rb
 config.view_component.preview_paths << "#{Rails.root}/spec/components/previews"
 ```
+
+If you need to test ViewComponents with behaviors taking POROs or ActiveRecord objects (even with unpersisted attributes) as input:
+
+```ruby
+module Helpers
+  module ViewComponentSystemSpecHelpers
+    def with_spec_preview(&preview_proc)
+      preview_class.define_method(:test_preview, &preview_proc)
+    end
+  end
+end
+
+class Post
+  attr_reader :title, :body
+
+  def initialize(title:, body:)
+    @title = title
+    @body = body
+  end
+end
+
+RSpec.describe PostComponent do
+  it "expands the body when clicking the button" do
+    post = Post.new(title: "My post title", body: "A long form text here")
+    with_spec_preview do
```

Comment:
Hmm, I'm hesitant to recommend this approach, as we're redefining `test_preview` each time this helper is used.

I think I'd prefer the approach @edwinthinks is working on in https://github.com/ViewComponent/view_component/pull/1061#pullrequestreview-1196421738. What do you think?

Reply:
Wow, thanks @joelhawksley for pointing that PR out. I totally missed it when looking for something solving the same problem. The solution proposed there @edwinthinks is indeed more thoughtful and robust, supporting a lot of customisations from the layout and slots standpoints.

> as we're redefining `test_preview` each time this helper is used.

As long as we don't have naming conflicts this shouldn't be an issue. I think it could be made better by defining a method with the example name, visiting the path and then undefining the method to avoid pollution or leaking.

Alas, seeing that work on #1061 is almost at the finish line, I'll keep this solution as a workaround in the application I'm working on until those helpers are released in a new version.